ksql-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -177,15 +180,20 @@ public SchemaKStream buildStream(final StreamsBuilder builder,
 
   }
 
-  private SchemaKTable tableForJoin(
+  // package private for test
+  SchemaKTable tableForJoin(
       final StreamsBuilder builder,
       final KsqlConfig ksqlConfig,
       final KafkaTopicClient kafkaTopicClient,
       final MetastoreUtil metastoreUtil,
       final FunctionRegistry functionRegistry,
       final Map<String, Object> props) {
 
-    final SchemaKStream schemaKStream = right.buildStream(builder, ksqlConfig, kafkaTopicClient, metastoreUtil, functionRegistry, props);
+    Map<String, Object> joinTableProps = new HashMap<>();
+    joinTableProps.putAll(props);
+    joinTableProps.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "EARLIEST");
```

Comment:
should "EARLIEST" be "earliest"?

Reply:
Yes, but in KSQL it's case insensitive. Regardless, I think it's better to use the correct one. Changed it to `earliest` :)